### PR TITLE
Move CF API internal endpoints to a different port

### PR DESCRIPTION
**Pull request by tcdowney**
##  WHAT is this change about?
We are moving the CF API internal endpoints to a separate port (one that is not exposed externally) and layering on some security mechanisms (`NetworkPolicy`, Istio `AuthorizationPolicy`) to protect them.

## Does this PR introduce a change to `config/values.yml`?
No

## Acceptance Steps
See linked story

### Expected
The call fails with a `403 Forbidden` error

### Actual
The call succeeds

## Tag your pair, your PM, and/or team
@cloudfoundry/cf-capi 

## Things to remember
- CAPI Story [#174455375](https://www.pivotaltracker.com/story/show/174455375)

### CODE REVIEW

File: config/networking/istio-authorization-policies.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#@ load("@ytt:data","data")
+
+#@ def principal(namespace, service_account):
+#@   return "cluster.local/ns/{}/sa/{}".format(namespace, service_account)
+#@ end
+
+#! https://istio.io/latest/docs/reference/config/security/authorization-policy/
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
```
Comment (tcdowney) [inline config/networking/istio-authorization-policies.yml:10]: @cloudfoundry/cf-networking 

What do y'all think about us introducing this `AuthorizationPolicy` use case?

File: config/networking/istio-authorization-policies.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+spec:
+  action: DENY
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: cf-api-server
+  rules:
+    - from:
+        - source:
+            notPrincipals:
+            - #@ principal(data.values.system_namespace, "eirini-events")
```
Comment (tcdowney) [inline config/networking/istio-authorization-policies.yml:32]: This limits our collaborators to pods with the `eirini-events` and `eirini-task-reporter` service accounts. This means that if those service accounts ever change we'll need to update this `AuthorizationPolicy`.

File: config/capi/_ytt_lib/capi-k8s-release/templates/api_server_deployment.yml [hunk 3]
```diff
@@ -77,11 +77,34 @@ spec:
           #@ if/end data.values.ccdb.ca_cert:
           - name: database-ca-cert
             mountPath: /config/database/certs
+        - name: package-image-uploader
```
Comment (tcdowney) [inline config/capi/_ytt_lib/capi-k8s-release/templates/api_server_deployment.yml:80]: This work is related to https://github.com/cloudfoundry/cf-for-k8s/issues/409. It's still in flight and isn't completely finished yet, we'll probably want to hold off on merging this (and autobumping capi-k8s-release) until that's complete as well.

Comment (tcdowney) [reply config/capi/_ytt_lib/capi-k8s-release/templates/api_server_deployment.yml:80]: Discussing this on Slack: https://cloudfoundry.slack.com/archives/C017LDM6KTQ/p1599088788032100

File: config/networking/istio-authorization-policies.yml (hunk 2 above)
Comment (ndhanushkodi) [reply config/networking/istio-authorization-policies.yml:32]: Will `action: DENY` + `notPrincipals` set up a double negative for that rule or you see it working? just a double check :)

File: config/networking/istio-authorization-policies.yml (hunk 1 above)
Comment (ndhanushkodi) [reply config/networking/istio-authorization-policies.yml:10]: seems ok

File: config/networking/istio-authorization-policies.yml (hunk 2 above)
Comment (tcdowney) [reply config/networking/istio-authorization-policies.yml:32]: I had trouble getting the `AuthorizationPolicy` to work as expected with `ALLOW`. Like it ended up only allowing traffic on the `/internal*` endpoints for the eirini principals and denying all other traffic. 😒 

So this one is set to `DENY` for everyone on the `/internal*` endpoints that is *not* in the list of principals specified. If that makes sense. I agree it's confusing.

I tested it by making the `cf-api-server` `NetworkPolicy` permissive and seeing that traffic was still denied to `/internal*` from a non-eirini Pod.

Comment (ndhanushkodi) [reply config/networking/istio-authorization-policies.yml:32]: Ah, I see, thank you!

**Comment by tcdowney** [comment]
We've completed the changes for https://github.com/cloudfoundry/cf-for-k8s/issues/409, we're going to open a new PR that more clearly has both those changes and these changes.

**Pull request closed without merging**